packages/core/src/browser/shell/application-shell.ts

```diff
@@ -744,6 +744,34 @@ export class ApplicationShell extends Widget {
         }
     }
 
+    /**
+     * Find the widget that contains the given HTML element. The returned widget may be one
+     * that is managed by the application shell, or one that is embedded in another widget and
+     * not directly managed by the shell, or a tab bar.
+     */
+    findWidgetForElement(element: HTMLElement): Widget | undefined {
```

Comment:
How about:
```ts
for (const widget of this.widgets) {
    if (widget.node.contains(element)) {
        return widget;
    }
}
```

Reply:
That would not pick up widgets that are not tracked by the shell, such as tab bars.